lib/base.js

```diff
@@ -250,9 +250,46 @@ class ConnectionPool extends EventEmitter {
           return
         }
 
+        let createAttempt = 0
+
+        function create () {
+          const createId = createAttempt++
+          let timeout = this.config.pool ? this.config.pool.acquireTimeoutMillis : undefined
+          let backoff
+          let retryCount = 0
+          const self = this
+          const run = function () {
+            debug('pool(%d): attempting to create connection resource(%d), attempt #%d', IDS.get(this), createId, retryCount)
+            return this._poolCreate().catch((e) => {
+              retryCount++
+              backoff = retryCount > 8 ? backoff : Math.pow(2, retryCount) * 100
+              if (typeof timeout !== 'undefined') {
+                timeout -= backoff
+                if (timeout <= 0) {
+                  throw e
+                }
+              }
+              return new Promise((resolve, reject) => {
+                const timer = setTimeout(() => {
+                  debug('pool(%d): backoff timer complete resource(%d)', IDS.get(self), createId)
+                  if (!self.connecting && !self.connected) {
+                    debug('pool(%d): pool closed while trying to acquire a connection resource(%d)', IDS.get(self), createId)
+                    reject(new ConnectionError('Connection is closed.', 'ECONNCLOSED'))
+                  } else {
+                    resolve()
+                  }
+                }, backoff)
+                timer.unref()
+                debug('pool(%d): failed to create connection resource(%d), retrying with backoff of %dms', IDS.get(self), createId, backoff)
+              }).then(run.bind(this))
```

Comment:
I don't think you need to `.bind` here because you're already referencing `self` (which could be renamed to something better)

Reply:
I'm mixing use of `self` and `this` which is a bit gross - `self` is mostly for the Promise function, though...

Reply:
`self` is a bit of a bad name, but the only sane name is `connectionPool` but that's pretty ambiguous, so I think I'm going to leave it as is

Reply:
In any case you need not re-bind.

Reply:
removing the `bind` breaks the behaviour